alf/algorithms/algorithm.py

```diff
@@ -1533,6 +1714,214 @@ def _update(self, experience, batch_info, weight):
+                     binfo,
+                     offline_batch,
+                     offline_binfo,
+                     weight=alf.nest.get_nest_size(offline_batch, 1) /
+                     mini_batch_size)
+                if do_summary:
+                    if exp:
+                        self.summarize_train(exp, train_info, loss_info,
+                                             params)
+                    # TODO: separate summarization
+                    self.summarize_train(offline_exp, offline_train_info,
+                                         offline_loss_info, None)
+
+        train_steps = 2 * batch_size * mini_batch_length * num_updates
+        return train_steps
+
+    def _hybrid_update(self, experience, batch_info, offline_experience,
+                       offline_batch_info, weight):
+        length = alf.nest.get_nest_size(offline_experience, dim=0)
+
+        if self._count == 0 or (
+            (self._count > self._rl_train_after_update_steps) and
+            (self._count % self._rl_train_every_update_steps == 0)):
+            # 1) RL training
+            RL_train = True
+            pre_train = False
+            if self._config.temporally_independent_train_step or length == 1:
+                train_info = self._collect_train_info_parallelly(experience)
+            else:
+                train_info = self._collect_train_info_sequentially(experience)
+
+            experience = dist_utils.params_to_distributions(
+                experience, self.processed_experience_spec)
+
+            loss_info = self.calc_loss(train_info)
+        else:
+            train_info = None
+            loss_info = None
+            RL_train = False
+            # some pre-training in the initial phase
+            pre_train = True
+
+        # 2) aux training
+        offline_train_info = self._collect_train_info_offline(
+            offline_experience, pre_train)
+
+        offline_loss_info = self.calc_loss_offline(offline_train_info,
+                                                   pre_train)
+
+        if loss_info is not None and loss_info.priority != ():
+            priority = (loss_info.priority**self._config.priority_replay_alpha
+                        + self._config.priority_replay_eps)
+            self._replay_buffer.update_priority(batch_info.env_ids,
+                                                batch_info.positions, priority)
+            if self._debug_summaries and alf.summary.should_record_summaries():
+                with alf.summary.scope("PriorityReplay"):
+                    summary_utils.add_mean_hist_summary(
+                        "new_priority", priority)
+                    summary_utils.add_mean_hist_summary(
+                        "old_importance_weight", batch_info.importance_weights)
+        else:
+            assert batch_info is None or batch_info.importance_weights == (), (
+                "Priority replay is enabled. But priority is not calculated.")
```

Comment:
Share this segment with those in _update()

Reply:
Changed to shared with update